### PR TITLE
DP-6346, DP-6416 restore the location filter in dataselections and filter

### PR DESCRIPTION
**Pull request by aditudorache**

### CODE REVIEW

File: src/shared/ducks/filters/filters.js [hunk 1]
```diff
@@ -48,6 +49,16 @@ const reducer = (state = {}, action) => {
       };
     }
 
+    case SET_GEOMETRY_FILTERS:
+      return {
+        ...state,
+        shape: action.payload.description === '' ? undefined : {
```
[the remark this answers is not in the document]
Comment (royderks) [reply src/shared/ducks/filters/filters.js]: Why not 'null'?

File: src/shared/sagas/data-selection/data-selection.js [hunk 2]
```diff
@@ -50,8 +51,16 @@ function* retrieveDataSelection(action) {
   } = action.payload;
   try {
     yield call(waitForAuthentication);
+
+    // exclude the geometryFilter from the attribute filters
```
[the remark this answers is not in the document]
Comment (royderks) [reply src/shared/sagas/data-selection/data-selection.js]: Can you add reference to DP-6442?

File: src/shared/ducks/filters/filters.js [hunk 3]
```diff
@@ -2,36 +2,36 @@ import { createSelector } from 'reselect';
 import { routing } from '../../../app/routes';
 import { getFilters as getDataSelectionFilters } from '../data-selection/selectors';
 import { getFilters as getDatasetFilters } from '../datasets/datasets';
+import { SET_GEOMETRY_FILTERS } from '../data-selection/constants';
```
Comment (chrisvanmook) [inline src/shared/ducks/filters/filters.js]: This is an anti pattern, we should never create a dependencies between reducers / ducks. Rather create a side-effect on SET_GEOMETRY_FILTERS (data-selection constant), that wil eventually trigger another SET_GEOMETRY_FILTERS (in the filter constants). This is also not the best approach (we should obviously refactor how this works) but for now, this is the least painful solution imho :)

Comment (aditudorache) [reply src/shared/ducks/filters/filters.js]: done

File: src/shared/sagas/data-selection/data-selection.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (aditudorache) [reply src/shared/sagas/data-selection/data-selection.js]: done

File: src/shared/ducks/filters/filters.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (aditudorache) [reply src/shared/ducks/filters/filters.js]: refactored

**Pull request merged**